solr/core/src/java/org/apache/solr/highlight/UnifiedSolrHighlighter.java

```diff
@@ -314,6 +343,24 @@ protected PassageFormatter getFormatter(String fieldName) {
       return new DefaultPassageFormatter(preTag, postTag, ellipsis, "html".equals(encoder));
     }
 
+    @Override
+    protected Comparator<Passage> getPassageSortComparator(String fieldName) {
+      PassageSort passageSort =
+          PassageSort.parse(
+              params.getFieldParam(fieldName, HighlightParams.PASSAGE_SORT, PassageSort.START_OFFSET.key));
+
+      switch (passageSort) {
+        case START_OFFSET:
+          return Comparator.comparingInt(Passage::getStartOffset);
+        case END_OFFSET:
+          return Comparator.comparingInt(Passage::getEndOffset);
+        case SCORE:
+          return Comparator.comparingDouble(Passage::getScore);
+        default:
+          throw new AssertionError();
```

Comment:
If a user supplies a bogus value, we should throw a SolrException with BAD_REQUEST as you can see elsewhere in this file.

Reply:
It was fixed in this commit: https://github.com/apache/solr/pull/2697/commits/deafe902701bf33013a5c694fc1e85e9c7530818.